loopy/codegen/control.py

```diff
@@ -472,9 +472,10 @@ def gen_code(inner_codegen_state):
                 prev_gen_code = gen_code
 
                 def gen_code(inner_codegen_state):  # pylint: disable=function-redefined
-                    condition_exprs = [
+                    condition_exprs = ([
                             constraint_to_cond_expr(cns)
-                            for cns in bounds_checks] + list(pred_checks)
+                            for cns in bounds_checks]
+                            + list(sorted(pred_checks, key=lambda pred: repr(pred))))
```

Comment:
Address linter failure, maybe also use use iterable unpacking.

Reply:
I've addressed the linting issue (wow ruff is fast!). I don't quite know what you mean about iterable unpacking.